rust/agama-lib/src/profile.rs

```diff
@@ -1,49 +1,34 @@
 use crate::error::ProfileError;
 use anyhow::Context;
-use curl::easy::Easy;
 use jsonschema::JSONSchema;
 use log::info;
 use serde_json;
 use std::{fs, io::Write, path::Path, process::Command};
 use tempfile::{tempdir, TempDir};
 use url::Url;
 
-/// Downloads a profile for a given location.
-pub struct ProfileReader {
+/// Downloads and converts autoyast profile.
+pub struct AutoyastProfile {
     url: Url,
 }
 
-impl ProfileReader {
-    pub fn new(url: &str) -> anyhow::Result<Self> {
-        let url = Url::parse(url)?;
-        Ok(Self { url })
+impl AutoyastProfile {
+    pub fn new(url: &Url) -> anyhow::Result<Self> {
+        Ok(Self { url: url.clone() })
     }
 
-    pub fn read(&self) -> anyhow::Result<String> {
+    pub fn read(&self, mut out_fd: impl Write) -> anyhow::Result<()> {
```

Comment:
np: I understand the reason, but looks weird that the function is called "read" and receives a "impl Write" argument :-)

Reply:
I can name it  `read_into`?

Reply:
I like it.